### DISCUSSION TITLE
Improving the app

### DISCUSSION THREAD

**Discussion by chetan-SMRI** [fel88/DeepNestPort#20] [Ideas]
firstly, I think we should rather update some features in the application itself,

1. Instead of showing values in pixel format we should rather show it in the inch format which is most used in this industry and is industry standard.
2. We should show sheets correctly(while running nesting process) , in the current application it is difficult to view sheets and zoom is also not possible, and also we cant scroll to see other sheets(we can copy code for this from Deepnestsharp)
3. DXF differentiator is needed, like in deepnest if u import any svg and if it has parts then they are automatically differentiated, but in your application they remain grouped as a single part.
After these important changes we can move to improve GUI .
Also tell me if u want to collab, I am a python developer but I have previously worked on C# so I will be able to help as much as I can do.

**Comment by chetan-SMRI** [comment]
also i got  to see one thing in the new core app, its GUI is nice but it only gives one nesting result instead of options, i guess u know this and correct this.

also anything like wastage, time saved, and other general info like how many parts of total placed is not visible

**Comment by fel88** [comment]
>   ...Instead of showing values in pixel format we  should rather show it in the inch format which is most used in this industry and is industry standard ...

I think I can add switcher between inch/mm (1px=1mm at the moment)

>   ... in the current application it is difficult to view sheets and zoom is also not possible ....
> 

It's really weird.. Have you tried scrolling with a mouse wheel?  doesn't it work?

>   DXF differentiator is needed, like in deepnest if u import any svg and if it has parts then they are automatically differentiated, but in your application they remain grouped as a single part.

It seems doable. I'll try to add a mini prompt dialog to choose the loading way (with or without splitting)

>    Also tell me if u want to collab, I am a python developer but I have previously worked on C# so I will be able to help as much as I can do.

Always glad of any help. :+1:

**Reply by chetan-SMRI** [reply]
> It's really weird.. Have you tried scrolling with a mouse wheel? doesn't it work?

I am talking about the latest core application, also I don't use mouse in my laptop so it will be more efficient if we add up a scroller,

Also in core application ui is good but setting is not visible, and we also cannot add new sheet or edit default sheet width height

**Reply by fel88** [reply]
I've fixed all the stuff a bit. 
Also I'll try to fix other problems ASAP

**Reply by chetan-SMRI** [reply]
it works good with the mouse but in laptop most users will not understand that we have to go to edge to scroll, so better if we add up one ourselves, or just add one controller like this so that it will be easy to move
![image](https://user-images.githubusercontent.com/84266613/224226382-ec8e8358-2f80-4975-9871-239802c9d952.png)

**Comment by chetan-SMRI** [comment]
I have a idea that seems doable by me(hardware,ie arduino and python and 3d modeling) and you(C# software side). 

I have a plan to do something like below
![mine](https://user-images.githubusercontent.com/84266613/224223313-b0d05b4d-23da-49e2-83e9-44a0f93259e9.png)

this will auto calculate the dxf for all the sheets and then comes the software play, we have to import all dxf and set parts within, 

It will be a time consuming process but it would be great if we do it. Also we can do premium , in my country most lazer machine guys will purchase this for their business.

I need your comments upon this. i also have two lazer machines so it will be easy to live experiment

**Comment by gaaban** [comment]
Optimization suggestions:
The more graphics there are, the more time-consuming it is. After analysis, it was found that every time NFP.cs adds SVG points, it will create a new set and then convert it into an array. Because there are too many graphics and SVG points, it consumes a lot of memory and time overall. Convert NFP After changing the Points type from SvgPoint [] array to List<SvgPoint>, efficiency has been significantly improved.

You can try it！

**Reply by fel88** [reply]
Hi, sounds reasonable, but anyway, could you provide PR ?
